frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx

```diff
@@ -0,0 +1,81 @@
+import { t } from "ttag";
+import { useCallback, useState } from "react";
+
+import { Text, Button, Group, Modal, Stack } from "metabase/ui";
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormGroupWidget,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import { ApiKeysApi } from "metabase/services";
+
+import { SecretKeyModal } from "./SecretKeyModal";
+
+export const CreateApiKeyModal = ({
+  onClose,
+  refreshList,
+}: {
+  onClose: () => void;
+  refreshList: () => void;
+}) => {
+  const [modal, setModal] = useState<"create" | "secretKey">("create");
+  const [secretKey, setSecretKey] = useState<string>("");
+
+  const handleSubmit = useCallback(
+    async vals => {
+      const response = await ApiKeysApi.create(vals);
+      setSecretKey(response.unmasked_key);
+      setModal("secretKey");
+      refreshList();
+    },
+    [refreshList],
+  );
+
+  if (modal === "secretKey") {
+    return <SecretKeyModal secretKey={secretKey} onClose={onClose} />;
+  }
+
+  if (modal === "create") {
+    return (
+      <Modal
+        size="30rem"
+        padding="xl"
+        opened
+        onClose={onClose}
+        title={t`Create a new API Key`}
+      >
+        <FormProvider initialValues={{}} onSubmit={handleSubmit}>
+          <Form data-testid="create-api-key-modal">
+            <Stack spacing="md">
+              <FormTextInput
+                name="name"
+                label={t`Key name`}
+                size="sm"
+                required
+              />
+              <FormGroupWidget
+                name="group_id"
+                label={t`Which group should this key belong to? The key will have the same permissions granted to that group.`}
+                size="sm"
+                required
+              />
```

Comment:
If I leave this field blank, the 'Create' button becomes disabled and looks like this:

<img width="300" alt="image" src="https://github.com/metabase/metabase/assets/130925/72abaccf-921e-462b-afbf-26286ccd34b7">

A few seconds later, the text changes back to 'Create' but the button is still disabled:

<img width="300" alt="image" src="https://github.com/metabase/metabase/assets/130925/c6f78206-5165-4baf-817f-1c1fb5476570">

Is this expected behavior?

Reply:
I added `Yup` Schema to these forms and added initialValues, which cleans up most of this.